hivemq-edge/src/frontend/src/extensions/datahub/components/nodes/CustomHandle.tsx

```diff
@@ -0,0 +1,52 @@
+import { CSSProperties, FC, HTMLAttributes, useMemo } from 'react'
+import { getConnectedEdges, Handle, HandleProps, useNodeId } from 'reactflow'
+
+import useDataHubDraftStore from '../../hooks/useDataHubDraftStore.ts'
+
+interface CustomHandleProps extends Omit<HandleProps & Omit<HTMLAttributes<HTMLDivElement>, 'id'>, 'isConnectable'> {
+  isConnectable?: boolean | number
+}
+
+export const CustomHandle: FC<CustomHandleProps> = (props) => {
+  const { nodes, edges } = useDataHubDraftStore()
+  const nodeId = useNodeId()
+
+  const isHandleConnectable = useMemo(() => {
+    if (typeof props.isConnectable === 'number') {
+      const node = nodes.find((e) => e.id === nodeId)
+      if (node) {
+        const connectedEdges = getConnectedEdges([node], edges)
+
+        const toHandle = connectedEdges.filter((e) => {
+          const otherEnd = props.type === 'source' ? e.sourceHandle : e.targetHandle
+          return otherEnd === props.id
+        })
```

Comment:
question: similar to the previous comment

>question(non-blocking): I'm a big fan of preventing abbreviations and use the sigular of a list
It becomes easier to parse. As an example I first thought it was an event then error then I saw it's a filter.

```suggestion
        const toHandle = connectedEdges.filter((edge) => {
          const otherEnd = props.type === 'source' ? edge.sourceHandle : edge.targetHandle
          return otherEnd === props.id
        })
```